core/crypto/crypto_store/key_file_storage.cpp

```diff
@@ -96,21 +96,13 @@ namespace kagome::crypto {
       KeyTypeId type,
       gsl::span<const uint8_t> public_key,
       gsl::span<const uint8_t> seed) const {
-    std::ofstream file;
-
     auto &&path = composeKeyPath(type, public_key);
-    file.open(path.native(), std::ios::out | std::ios::trunc);
-    if (!file.is_open()) {
-      return Error::FAILED_OPEN_FILE;
-    }
-    auto hex = common::hex_lower(seed);
+    OUTCOME_TRY(saveKeyHexAtPath(seed, path));
     SL_TRACE(logger_,
              "Saving keypair (public: {}, secret: {}) to {}",
              common::hex_lower(public_key),
-             hex,
+             common::hex_lower(seed),
```

Comment:
ATTENTION: Please, avoid logging of secret data in any case.

Reply:
Additionaly it is best practice to check if file is allowed only for current user (file access mode must be 600 aka -rw-------), and create new file with the same mode.